src/OpenTelemetry.AutoInstrumentation.Loader/Loader.Net.cs

```diff
@@ -21,6 +23,29 @@ private static string ResolveManagedProfilerDirectory()
         return Path.Combine(tracerHomeDirectory, tracerFrameworkDirectory);
     }
 
+    private static string[]? GetStoreFiles()
+    {
+        try
+        {
+            var storeDirectory = Environment.GetEnvironmentVariable("DOTNET_SHARED_STORE");
+            if (storeDirectory == null || !Directory.Exists(storeDirectory))
+            {
+                return null;
+            }
+
+            var architecture = Environment.Is64BitProcess ? "x64" : "x86";
```

Comment:
Do we need to handle arm64 here too?

Reply:
arm64 experimental support, and I'm not sure if we need it. This is a good point and needs to be explored further as a follow-up.

Reply:
Changes for arm64 was very small and I could cover in this PR.
Diagnostic rule has a similar code and arm64 was not updated in it. This should be followed up in a different PR as it is not related to this issue.

Reply:
Fixed f2410d6d882fa05040e0319ba66e4d3e801b5ed5